api/client/stats_unit_test.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"sync"
+	"testing"
+)
+
+type TestWriter struct {
```

Comment:
I think you could use a [`bytes.Buffer`](http://golang.org/pkg/bytes/#Buffer) instead of a custom type.